packages/jats-fetch/src/download.ts

```diff
@@ -101,54 +118,56 @@ async function getJatsUrlFromDoi(
 /**
  * Attempt to download JATS from provided input
  *
- * `urlOrDoi` may be (1) a local file, in which case, the file content is
+ * `urlOrId` may be (1) a local file, in which case, the file content is
  * directly returned, (2) PubMed ID, PubMed Central ID, or DOI, in which case,
  * possible download links are constructed and followed, or (3) a direct
  * download URL, in which case, the content is fetched.
  */
 export async function downloadJatsFromUrl(
   session: ISession,
-  urlOrDoi: string,
+  urlOrId: string,
   opts: ResolutionOptions = {},
 ): Promise<DownloadResult> {
-  if (fs.existsSync(urlOrDoi)) {
-    session.log.debug(`JATS returned from local file ${urlOrDoi}`);
-    const data = fs.readFileSync(urlOrDoi).toString();
-    return { success: true, source: urlOrDoi, data };
+  let result: DownloadResult = { success: false, source: urlOrId };
+  if (fs.existsSync(urlOrId)) {
+    session.log.debug(`JATS returned from local file ${urlOrId}`);
+    const data = fs.readFileSync(urlOrId).toString();
+    return { success: true, source: urlOrId, data };
   }
-  const expectedUrls: string[] = [];
-  try {
-    // Custom resolvers are prioritized; in these cases we will know more about the "flavor" of the JATS
-    expectedUrls.push(await customResolveJatsUrlFromDoi(session, urlOrDoi, opts));
-  } catch {
-    session.log.debug(`No custom resolvers match ${urlOrDoi}`);
-  }
-  expectedUrls.push(
-    ...(
-      await Promise.all([
-        constructJatsUrlFromPubMedCentral(session, urlOrDoi, opts),
-        getJatsUrlFromDoi(session, urlOrDoi, opts),
-      ])
-    ).filter((u): u is string => !!u),
-  );
-  if (isUrl(urlOrDoi)) {
-    expectedUrls.push(urlOrDoi);
-  }
-  if (expectedUrls.length > 0) {
-    session.log.debug(['Trying URLs:\n', ...expectedUrls.map((url) => ` ${url}\n`)].join('  - '));
-    for (let index = 0; index < expectedUrls.length; index++) {
-      const url = expectedUrls[index];
-      try {
-        const data = await downloadFromUrl(session, url, opts);
-        if (data) return { success: true, source: url, data };
-      } catch (error) {
-        session.log.debug((error as Error).message);
-      }
+  if (doi.validate(urlOrId)) {
```

Comment:
This function I split up to make the doi vs pmc vs other paths clearer.

This appears to reduce some parallelism, but I don't think it actually makes much of a difference, since the original parallel functions were often failing fast based on "is doi" vs "is pmc" checks.